File: src/test/java/com/jocoos/mybeautip/domain/file/api/front/FileControllerTest.java
package com.jocoos.mybeautip.domain.file.api.front;

import com.jocoos.mybeautip.global.config.restdoc.RestDocsTestSupport;
import org.junit.jupiter.api.Test;
import org.springframework.mock.web.MockMultipartFile;
import org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders;
import org.springframework.restdocs.payload.JsonFieldType;
import org.springframework.security.test.context.support.WithUserDetails;
import org.springframework.test.web.servlet.ResultActions;

import static org.junit.jupiter.api.Assertions.*;
import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.document;
import static org.springframework.restdocs.payload.PayloadDocumentation.fieldWithPath;
import static org.springframework.restdocs.payload.PayloadDocumentation.responseFields;
import static org.springframework.restdocs.request.RequestDocumentation.partWithName;
import static org.springframework.restdocs.request.RequestDocumentation.requestParts;
import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

class FileControllerTest extends RestDocsTestSupport {


    @Test
    @WithUserDetails(value = "4", userDetailsServiceBeanName = "mybeautipUserDetailsService")
    void uploadFiles() throws Exception {
        MockMultipartFile file = new MockMultipartFile("files", "file", "image/jpeg", "mock".getBytes());
        ResultActions result = mockMvc.perform(RestDocumentationRequestBuilders
                        .multipart("/api/1/file")
                        .file(file))
                .andExpect(status().isOk())
                .andDo(print());

        result.andDo(document("upload_file",
                        requestParts(
                                partWithName("files").description("업로드할 파일 목록")
                        ),
                        responseFields(
                                fieldWithPath("[]").type(JsonFieldType.ARRAY).description("UPLOAD 된 파일 URL")
                        )
                )
        );
    }
}

File: src/test/java/com/jocoos/mybeautip/domain/file/api/front/FileControllerTest.kt
package com.jocoos.mybeautip.domain.file.api.front

import com.jocoos.mybeautip.global.config.restdoc.RestDocsIntegrationTestSupport
import org.junit.jupiter.api.Test
import org.springframework.http.HttpHeaders.AUTHORIZATION
import org.springframework.mock.web.MockMultipartFile
import org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.document
import org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.multipart
import org.springframework.restdocs.payload.JsonFieldType.ARRAY
import org.springframework.restdocs.payload.PayloadDocumentation.fieldWithPath
import org.springframework.restdocs.payload.PayloadDocumentation.responseFields
import org.springframework.restdocs.request.RequestDocumentation.partWithName
import org.springframework.restdocs.request.RequestDocumentation.requestParts
import org.springframework.test.web.servlet.ResultActions
import org.springframework.test.web.servlet.result.MockMvcResultHandlers.print
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

class FileControllerTest(

) : RestDocsIntegrationTestSupport() {

    @Test
    fun uploadFiles() {
        val file = MockMultipartFile("files", "file", "image/jpeg", "mock".toByteArray())

        val result: ResultActions = mockMvc
            .perform(
            multipart("/api/1/file")
                .file(file)
                .header(AUTHORIZATION, requestUserToken)
        )
            .andExpect(status().isOk)
            .andDo(print())
        result.andDo(
            document(
                "upload_file",
                requestParts(
                    partWithName("files").description("업로드할 파일 목록")
                ),
                responseFields(
                    fieldWithPath("[]").type(ARRAY).description("UPLOAD 된 파일 URL")
                )
            )
        )
    }
}
